agent/src/main/java/io/pyroscope/javaagent/PyroscopeAgent.java

```diff
@@ -35,22 +37,40 @@ public static void start(Config config) {
 
     public static void start(Options options) {
         Logger logger = options.logger;
-
+       if( !startOptions.compareAndSet(null, options)) {
+        logger.log(Logger.Level.ERROR, "Failed to start profiling - already started");
+        return;
+        }
         if (!options.config.agentEnabled) {
             logger.log(Logger.Level.INFO, "Pyroscope agent start disabled by configuration");
             return;
         }
+        logger.log(Logger.Level.DEBUG, "Config: %s", options.config);
+        try {
+            startOptions.compareAndSet(null, options);
+            startOptions.get().scheduler.start(startOptions.get().profiler);
+            logger.log(Logger.Level.INFO, "Profiling started");
+        } catch (final Throwable e) {
+            logger.log(Logger.Level.ERROR, "Error starting profiler %s", e);
+        }
+    }
 
-        if (!started.compareAndSet(false, true)) {
+    /**
+     * stop is used to stop profiling
+     */
+    public static void stop() {
+        Logger logger = startOptions.get().logger;
```

Comment:
This will crash when startOptions is null

Reply:
Added a null check